### PR TITLE
feat(Tooltip): open on hover in desktop

### PR DESCRIPTION
**Pull request by sergivillar**

### CODE REVIEW

**Review by atabel — COMMENTED** [review]
In desktop, I'd apply a timeout when the user hovers the target until the tooltip is shown.

Please include someone from Design Core to review these behavior changes

**Comment by sergivillar** [comment]
> In desktop, I'd apply a timeout when the user hovers the target until the tooltip is shown.
> 
> Please include someone from Design Core to review these behavior changes

@HectorSancho told me to avoid that delay, but let's wait for his opinion

**Review by pladaria — APPROVED** [review]

**Comment by yceballost** [comment]
missing slot feature in tooltip
image

File: src/tooltip.tsx [hunk 1]
```diff
@@ -97,12 +104,152 @@ const useStyles = createUseStyles((theme) => {
     };
```
[the remark this answers is not in the document]
Comment (sergivillar) [reply src/tooltip.tsx:104]: by default the width is 340px in desktop, do you want to change that?

[the remark this answers is not in the document]
Comment (yceballost) [reply src/tooltip.tsx:104]: in desktop would be adapted to the content width up to size 343px
image

File: src/tooltip.tsx [hunk 2]
```diff
@@ -282,10 +434,10 @@ const Tooltip: React.FC<Props> = ({children, description, target, title, targetL
                                 {description && <p className={classes.description}>{description}</p>}
```
Comment (yceballost) [inline src/tooltip.tsx:469]: tooltip have to use text-preset-2/regular instead of `<p>`

File: src/__stories__/tooltip-story.tsx [hunk 3]
```diff
@@ -65,9 +65,14 @@ export const Default: StoryComponent = () => {
                     target={<Text2 regular>Tooltip with link</Text2>}
                     position="top"
                     description={description}
-                >
-                    <a href="#whatever">Ir a la web</a>
-                </Tooltip>
+                    extra={<a href="#whatever">Ir a la web</a>}
```
Comment (yceballost) [inline src/__stories__/tooltip-story.tsx:68]: better use textLink for this example 👀

**Review by yceballost — APPROVED** [review]
nice!! 😍

File: src/tooltip.tsx [hunk 4]
```diff
@@ -478,12 +454,12 @@ const Tooltip: React.FC<Props> = ({children, description, target, title, targetL
...
-                                {title && <p className={classes.title}>{title}</p>}
-                                {description && <p className={classes.description}>{description}</p>}
-                            </>
+                            <Stack space={4}>
+                                {title && <Text2 medium>{title}</Text2>}
+                                {description && <Text2 regular>{description}</Text2>}
+                            </Stack>
                         )}
-                        {children}
+                        {extra || children}
```
Comment (atabel) [inline src/tooltip.tsx:462]: why do we need `extra`? just use `children`

File: src/tooltip.tsx [hunk 5]
```diff
@@ -75,48 +85,150 @@ const useStyles = createUseStyles((theme) => {
...
 
-const Tooltip: React.FC<Props> = ({children, description, target, title, targetLabel, ...rest}) => {
+const Tooltip: React.FC<Props> = ({
+    children,
+    extra,
+    description,
+    target,
+    title,
+    targetLabel,
+    delay = 0,
```
Comment (atabel) [inline src/tooltip.tsx:222]: isn't `delay` a boolean?
it should have delay by default

File: src/tooltip.tsx [hunk 6]
```diff
@@ -1,20 +1,30 @@
...
+const distanceToTarget = 4 + arrowSize;
 const marginLeftRightMobile = 16;
 const defaultWidthDesktop = 340;
 const arrowWrapperWidth = arrowSize * 2;
 const arrowWrapperHeight = arrowSize;
 
+const transitionDurationMs = 500;
+const animationMovement = 12;
+const animationTiming = 'cubic-bezier(0.215, 0.61, 0.355, 1)';
+const defaultShowTooltipDelayMs = 2000;
```
Comment (atabel) [inline src/tooltip.tsx:24]: too much, I'd say something like 500ms should be fine

File: src/tooltip.tsx (hunk 4 above)
Comment (yceballost) [reply src/tooltip.tsx:462]: looking for feature parity

File: src/tooltip.tsx (hunk 6 above)
Comment (yceballost) [reply src/tooltip.tsx:24]: we can change it to 500ms

But when you hover some elements for example, PR Chip in this page, the delay is more than 500ms, I think is a native delay, just comment
image

File: src/tooltip.tsx (hunk 4 above)
Comment (atabel) [reply src/tooltip.tsx:462]: it doesn't make sense to have both, if we add extra we should remove children. But removing children is a breaking change, so if we prefer extra, then mark children as deprecated

**Review by atabel — APPROVED** [review]

File: src/__stories__/tooltip-story.tsx [hunk 7]
```diff
@@ -62,12 +62,26 @@ export const Default: StoryComponent = () => {
...
                     position="top"
                     description={description}
-                >
-                    <a href="#whatever">Ir a la web</a>
-                </Tooltip>
+                    extra={<Placeholder />}
+                />
+                <Tooltip
+                    targetLabel="help text"
+                    target={<Text2 regular>Tooltip with delay</Text2>}
```
Comment (atabel) [inline src/__stories__/tooltip-story.tsx:81]: does this example makes sense? all the Tooltips have delay now

Comment (sergivillar) [reply src/__stories__/tooltip-story.tsx]: changed

**Pull request merged**